### PR TITLE
Update WalletOptions + some Clippy

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Update WalletOptions for single account wallet + some Clippy 

## Links to any relevant issues

Related to #1424 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: bindings/core/src/lib.rs [hunk 1]
```diff
@@ -42,14 +43,21 @@ pub fn init_logger(config: String) -> std::result::Result<(), fern_logger::Error
...
     pub secret_manager: Option<SecretManagerDto>,
+    pub storage_path: Option<String>,
 }
 
 impl WalletOptions {
+    pub fn with_address(mut self, address: impl Into<Option<Bech32Address>>) -> Self {
+        self.address = address.into();
+        self
+    }
+
```
Comment (Alex6323) [inline bindings/core/src/lib.rs]: why no `with_alias`?

Comment (Alex6323) [reply bindings/core/src/lib.rs]: also ... a bit nit-pickish ... but can you align the method order with the field order?

File: bindings/core/src/lib.rs [hunk 2]
```diff
@@ -76,6 +84,15 @@ impl WalletOptions {
...
 
+        if let Some(address) = self.address {
+            builder = builder.with_address(address);
+        }
+        if let Some(alias) = self.alias {
+            builder = builder.with_alias(alias);
+        }
+        if let Some(bip_path) = self.bip_path {
+            builder = builder.with_bip_path(bip_path);
+        }
```
Comment (Alex6323) [inline bindings/core/src/lib.rs]: isn't that already handled in L85?

**Review by kwek20 — DISMISSED** [review]

**Review by kwek20 — COMMENTED** [review]
alex was better, unapproved from me :)

File: bindings/core/src/lib.rs (hunk 1 above)
Comment (Thoralf-M) [reply bindings/core/src/lib.rs]: Wondering if we should remove them all, they're not used apart from tests
But added and aligned

File: bindings/core/src/lib.rs (hunk 2 above)
Comment (Thoralf-M) [reply bindings/core/src/lib.rs]: Good catch

**Review by kwek20 — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**